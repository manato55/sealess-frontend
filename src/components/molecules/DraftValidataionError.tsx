import React,{useEffect,useState} from 'react'
import ErrorMessageWrapper from '../atoms/ErrorMessageWrapper'
import { useRecoilValue, useRecoilState } from 'recoil'
import { eachErrorFlag, authErrorMessage } from '../../store/atom'


interface Props {
    
}



export const DraftValidataionError = (props: Props) => {
    const [fileNumber, setFileNumber] = useState<string[]>([])
    const [errorFlag, setErrorFlag] = useRecoilState(eachErrorFlag)
    const errorMessage = useRecoilValue(authErrorMessage)

    useEffect(() => {
        return () => {
            setErrorFlag({...errorFlag, file: false, title:false, content: false, route: false})
            setFileNumber([])
        }
    }, []);

    useEffect(() => {
        // 添付したファイル数に応じてエラーを表示する
        if(errorFlag.file) {
            let keys: string[] = Object.keys(errorMessage)
            let fileExtracted: string[] = keys.filter(v => v.match(/file/))
            setFileNumber(fileExtracted)
        }
    }, [errorFlag])

    return (
        <>
            <ErrorMessageWrapper>{errorFlag.title && errorMessage.title[0]}</ErrorMessageWrapper>
            <ErrorMessageWrapper>{errorFlag.content && errorMessage.content[0]}</ErrorMessageWrapper>
            <ErrorMessageWrapper>{errorFlag.route && errorMessage.route[0]}</ErrorMessageWrapper>
            {errorFlag.file && fileNumber.map((file,index) => 
                <ErrorMessageWrapper key={index}>{errorMessage[file][0]}</ErrorMessageWrapper>
            )}
        </>
    )
}

export default DraftValidataionError