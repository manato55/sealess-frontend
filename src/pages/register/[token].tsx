import RegisterToken from '../../components/organisms/RegisterToken';

const Token = () => {
  return (
    <>
      <RegisterToken />
    </>
  );
};

export default Token;
