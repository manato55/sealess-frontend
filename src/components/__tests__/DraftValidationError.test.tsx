import React from 'react';
import DraftValidataionError from '../molecules/DraftValidataionError';
import 'jsdom-global/register';
import renderer from 'react-test-renderer';
import { RecoilRoot } from 'recoil';

describe('DraftValidataionError test', () => {
  test('snapshot', () => {
    const tree = renderer
      .create(
        <RecoilRoot>
          <DraftValidataionError />
        </RecoilRoot>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
