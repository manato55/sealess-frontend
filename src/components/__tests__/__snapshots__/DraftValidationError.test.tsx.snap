// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraftValidataionError test snapshot 1`] = `
Array [
  <p
    className="sc-bdnxRM jbauqE"
  />,
  <p
    className="sc-bdnxRM jbauqE"
  />,
  <p
    className="sc-bdnxRM jbauqE"
  />,
]
`;
